src/components/stepforms/ColumnPicker.vue

```diff
@@ -46,38 +44,36 @@ export default class ColumnPicker extends Vue {
   @Prop({ default: true })
   syncWithSelectedColumn!: boolean;
 
-  // Only manage the deletion of 1 column at once at this stage
-  column: string | null = null;
-
   @VQBModule.Mutation setSelectedColumns!: MutationCallbacks['setSelectedColumns'];
   @VQBModule.Getter selectedColumns!: string[];
   @VQBModule.Getter columnNames!: string[];
 
   created() {
-    if (this.value) {
-      this.column = this.value;
-    } else {
-      const selected = this.selectedColumns;
-      if (selected.length) {
-        this.column = selected[0];
-        this.valueChanged();
-      }
+    if (this.syncWithSelectedColumn && this.selectedColumns[0] && !this.value) {
+      this.updateColumn(this.selectedColumns[0]);
     }
   }
 
-  valueChanged() {
-    if (this.column) {
-      // make sure to emit @input so that v-model in hosting component is notified
-      this.$emit('input', this.column);
-      this.setSelectedColumns({ column: this.column });
+  @Watch('selectedColumns')
+  onSelectedColumnsChanged() {
+    if (
+      this.syncWithSelectedColumn &&
+      this.selectedColumns[0] &&
+      this.selectedColumns[0] !== this.value
+    ) {
+      this.updateColumn(this.selectedColumns[0]);
     }
   }
 
-  @Watch('selectedColumns')
-  onSelectedColumnsChanged(val: string[], oldVal: string[]) {
-    if (!_.isEqual(val, oldVal) && this.syncWithSelectedColumn) {
-      this.column = val[0];
+  valueChanged(newColumn: string) {
+    this.updateColumn(newColumn);
```

Comment:
```suggestion
    this.$emit('input', newColumn);
```
is more explicit IMO...